cluster/gce/windows/common.psm1

```diff
@@ -146,6 +146,307 @@ function MustDownload-File {
+  # you must check both the return value of the function and the error code returned by GetLastError to
+  # determine whether or not an error has occurred. If an error has occurred, the return value of SetFilePointer
+  # is INVALID_SET_FILE_POINTER and GetLastError returns a value other than NO_ERROR.
+  $ret = $Kernel32::SetFilePointer($handle, 0, [System.IntPtr]::Zero, $FILE_BEGIN)
+  $err = [System.Runtime.InteropServices.Marshal]::GetLastWin32Error()
+  if ($ret -eq $INVALID_SET_FILE_POINTER -and $err -ne $NO_ERROR) {
+    Close-Handle -Handle $handle
+    throw "Failed to set file pointer for handle ${handle}, system error code: ${err}"
+  }
+
+  $ret = $Kernel32::SetEndOfFile($handle)
+  $err = [System.Runtime.InteropServices.Marshal]::GetLastWin32Error()
+  if ($ret -eq 0) {
+    Close-Handle -Handle $handle
+    throw "Failed to set end of file for handle ${handle}, system error code: ${err}"
+  }
+  Close-Handle -Handle $handle
+}
+
+class FileRotationConfig {
+  # Force rotation, ignoring $MaxBackupInterval and $MaxSize criteria.
+  [bool]$Force
+  # Maximum time since last backup, after which file will be rotated.
+  # When no backups exist, Rotate-File acts as if -MaxBackupInterval has not elapsed,
+  # instead relying on the other criteria.
+  [TimeSpan]$MaxBackupInterval
+  # Maximim file size, after which file will be rotated.
+  [int]$MaxSize
+  # Maximim number of backup archives to maintain.
+  [int]$MaxBackups
+}
+
+function New-FileRotationConfig {
+  param (
+    # Force rotation, ignoring $MaxBackupInterval and $MaxSize criteria.
+    [parameter(Mandatory=$false)] [switch]$Force,
+    # Maximum time since last backup, after which file will be rotated.
+    # When no backups exist, Rotate-File acts as if -MaxBackupInterval has not elapsed,
+    # instead relying on the other criteria.
+    # Defaults to weekly rotations.
+    [parameter(Mandatory=$false)] [TimeSpan]$MaxBackupInterval = $(New-TimeSpan -Day 7),
+    # Maximim file size, after which file will be rotated.
+    [parameter(Mandatory=$false)] [int]$MaxSize = 100mb,
+    # Maximim number of backup archives to maintain.
+    [parameter(Mandatory=$false)] [int]$MaxBackups = 5
+  )
+  $config = [FileRotationConfig]::new()
+  $config.Force = $Force
+  $config.MaxBackupInterval = $MaxBackupInterval
+  $config.MaxSize = $MaxSize
+  $config.MaxBackups = $MaxBackups
+  return $config
+}
+
+# Returns a list of paths to backup files for the original file path -Path,
+# assuming that backup files are in the same directory, with a prefix matching
+# the original file name and a .zip suffix.
+function Get-Backups {
+  param (
+    # Original path of the file for which backups were created (no suffix).
+    [parameter(Mandatory=$true)] [string]$Path
+  )
+  $parent = Split-Path -Parent -Path $Path
+  $leaf = Split-Path -Leaf -Path $Path
+  $files = Get-ChildItem -File -Path $parent |
+           Where-Object { $_.Name -like "${leaf}*.zip" }
```

Comment:
FYI, I think this can be just 'Where-Object Name -Like "${leaf}*.zip'; no $_ or braces required.

Reply:
Ok, let met test that.